src/components/TransactionHistoryRows/TransactionGenericRow.tsx

```diff
@@ -14,6 +14,26 @@ import { fromBaseUnit } from 'lib/math'
 import { TxId } from 'state/slices/txHistorySlice/txHistorySlice'
 import { breakpoints } from 'theme/theme'
 
+export const GetTxLayoutFormats = ({ parentWidth }: { parentWidth: number }) => {
+  const isLargerThanSm = parentWidth > parseInt(breakpoints['sm'], 10)
+  const isLargerThanMd = parentWidth > parseInt(breakpoints['md'], 10)
+  const isLargerThanLg = parentWidth > parseInt(breakpoints['lg'], 10)
+  let columns = '1fr'
+  let dateFormat = 'MM/DD/YYYY hh:mm A'
+
+  if (isLargerThanSm) {
+    columns = '1fr 2fr'
+    dateFormat = 'hh:mm A'
+  }
+  if (isLargerThanMd) {
+    columns = '1fr 2fr'
+  }
+  if (isLargerThanLg) {
+    columns = '1fr 2fr 1fr 1fr'
+  }
+  return { columns, dateFormat, breakPoints: [isLargerThanLg, isLargerThanMd, isLargerThanSm] }
+}
+
```

Comment:
I'm no expert but this feels like a pretty heavy handed approach to formatting based on screen size.

Do [the docs](https://chakra-ui.com/docs/styled-system/features/responsive-styles) offer any suggestions for simpler approaches?

Reply:
This isn’t based on screen size. It’s based on the size of the parent container. Unfortunately there is no real way to do this via css we only have access to the viewport size.

This gives us total control of the way the row looks at any size. Reducing the need for several versions of this component.